module/documents/activity/cast.mjs

```diff
@@ -141,15 +141,31 @@ export default class CastActivity extends ActivityMixin(CastActivityData) {
   getSpellChanges() {
     const changes = [];
     const source = this.toObject();
+
+    // Override spell details
     for ( const type of ["activation", "duration", "range", "target"] ) {
       if ( !this[type].override ) continue;
       const data = source[type];
       delete data.override;
       changes.push({ key: `system.${type}`, mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE, value: JSON.stringify(data) });
     }
+
+    // Remove ignored properties
     for ( const property of this.spell.properties ) {
       changes.push({ key: "system.properties", mode: CONST.ACTIVE_EFFECT_MODES.ADD, value: `-${property}` });
     }
+
+    // Set challenge overrides
+    const challenge = this.spell.challenge;
+    if ( challenge.override && challenge.attack ) changes.push(
+      { key: "activity.attack.attack.bonus", mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE, value: challenge.attack },
+      { key: "activity.attack.attack.flat", mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE, value: true }
+    );
+    if ( challenge.override && challenge.save ) changes.push(
+      { key: "activity.save.save.dc.calculation", mode: CONST.ACTIVE_EFFECT_MODES.OVERRIDE, value: "" },
```

Comment:
Since this is a 'special' key that doesn't operate as normal, I wonder if we want to indicate that in a way that's a bit more explicit. I was thinking of maybe using a sigil like `$activity` or `!activity`, but I would like to see more use-cases for these sorts of special keys before committing to something like that. Also, there are some special sigils used for serialisation in various parts of the core API that might complicate things.

For now maybe just a bit more explicit like `activitiesOfType.attack...`, `activitiesOfType.save...`, etc.? What do you think?

Reply:
Hmm, we could do something like `activities[attack].attack…` perhaps, to keep a clear separation between the types and the following key path.